File: data/code_pairs/cpp/2330.cpp
#include <vector>
#include <algorithm>

void b4step2(
    int mbc, int mx, int my, int meqn,
    std::vector<double>& q, double xlower, double ylower, 
    double dx, double dy, double t, double dt, 
    int maux, std::vector<double>& aux)
{
    if (t > 1.0) {
        for (int k = 0; k < maux; ++k) {
            if (k == 12) { // Adjusted for 0-based indexing
                for (int j = 0; j < my + 2 * mbc; ++j) {
                    for (int i = 0; i < mx + 2 * mbc; ++i) {
                        aux[k * (mx + 2 * mbc) * (my + 2 * mbc) + j * (mx + 2 * mbc) + i] = 0.0;
                    }
                }
            }
        }
    }
}

File: data/code_pairs/fortran/2330.f90
subroutine b4step2(mbc, mx, my, meqn, q, xlower, ylower, dx, dy, t, dt, maux, aux)
    implicit none
    integer, intent(in) :: mbc, mx, my, meqn, maux
    real(kind=8), intent(in) :: xlower, ylower, dx, dy, t, dt
    real(kind=8), intent(inout) :: q(meqn, 1-mbc:mx+mbc, 1-mbc:my+mbc)
    real(kind=8), intent(inout) :: aux(maux, 1-mbc:mx+mbc, 1-mbc:my+mbc)

    if (t > 1.d0) then
        aux(13, :, :) = 0.d0
    endif

end subroutine b4step2